ebpfsvc/rpc_api.cpp

```diff
@@ -5,29 +5,38 @@
 #include <stdexcept>
 #include <stdio.h>
 #include <vector>
-#include "api_internal.h"
 #include "api_service.h"
-#include "ebpf_windows.h"
 #include "rpc_interface_h.h"
 #include "svc_common.h"
-#include "Verifier.h"
 
 // Critical section to serialize RPC calls.
 // Currently ebpfsvc uses a global context to track verification
 // and JIT compilation, hence all RPC calls should be serialized.
 static std::mutex _mutex;
 
 ebpf_result_t
-ebpf_server_verify_and_jit_program(
+ebpf_server_verify_and_load_program(
     /* [ref][in] */ ebpf_program_load_info* info,
     /* [ref][out] */ uint32_t* logs_size,
     /* [ref][size_is][size_is][out] */ char** logs)
 {
-    UNREFERENCED_PARAMETER(info);
-    UNREFERENCED_PARAMETER(logs_size);
-    UNREFERENCED_PARAMETER(logs);
+    if (info->byte_code_size == 0) {
+        return EBPF_INVALID_ARGUMENT;
+    }
+
+    std::scoped_lock lock(_mutex);
```

Comment:
Is this still needed? I thought the change to thread local for state avoided this now?

Reply:
This is not needed now. Removed.